src/Amalgam/AssetManager.cpp

```diff
@@ -401,30 +553,30 @@ Entity *AssetManager::LoadEntityFromResource(AssetParameters &asset_params, bool
 	}
 
 	if(persistent)
-		SetEntityPersistence(new_entity, &asset_params);
+		SetEntityPersistence(new_entity, asset_params);
 
 	//load contained entities
 
 	//iterate over all files in directory
-	std::string contained_entities_directory = asset_params.resourceBasePath + "/";
+	std::string contained_entities_directory = asset_params->resourceBasePath + "/";
```

Comment:
If we're on C++17 or later, can we use [`std::filesystem::path`](https://en.cppreference.com/w/cpp/filesystem/path) here?